misc/toml-parser/src/main/java/sementic/nodes/SemanticAnalyzer.java

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package sementic.nodes;
+
+import sementic.tools.DiagnosticCode;
+
+import java.util.Map;
+
+/**
+ * Semantic Analyzer for TOML.
+ */
+public class SemanticAnalyzer extends TomlNodeVisitor {
```

Comment:
I think we should call this `TOMLValidator`, since it technically doesn't validate semantics (e.g: doesn't do type checking etc.)